be/src/exec/olap_scan_node.cpp

```diff
@@ -1368,9 +1375,14 @@ void OlapScanNode::scanner_thread(OlapScanner* scanner) {
     cpu_watch.start();
     Status status = Status::OK();
     bool eos = false;
+    if (UNLIKELY(_transfer_done)) {
+        eos = true;
+        status = Status::Cancelled("Cancelled");
+        LOG(INFO) << "Scan thread cancelled, cause query done, scan thread started to exit";
+    } 
     RuntimeState* state = scanner->runtime_state();
     DCHECK(NULL != state);
-    if (!scanner->is_open()) {
+    if (!eos && !scanner->is_open()) {
```

Comment:
If scanner is not opened, we need to take care of "close()" method

Reply:
Modified to when _transfer_done` is True, the scan thread will exit directly after starting, 
and no longer execute the subsequent logic including the scanner's `close()` method.